### PR TITLE
Ssrc rewriting

### PR DESCRIPTION
**Pull request by jqdrqgnq**

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -303,6 +305,15 @@ class ConferenceConnector {
             this._handleConferenceJoined.bind(this));
         room.on(JitsiConferenceEvents.CONFERENCE_FAILED,
             this._onConferenceFailed.bind(this));
+        room.on(JitsiConferenceEvents.TRACK_OWNER_CHANGED, (ssrc, owner) => {
```
Comment (jallamsetty1) [inline conference.js:308]: We need to add an event listener [here](https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/conference/actions.js#L96) as well for ssrc-rewriting to work on react-native clients.

Comment (jqdrqgnq) [reply conference.js:308]: done

**Comment by jallamsetty1** [comment]
Jenkins test. this please.

File: react/features/base/config/functions.any.ts [hunk 2]
```diff
@@ -85,6 +85,16 @@ export function getSourceNameSignalingFeatureFlag(state: IState) {
...
 }
 
+/**
+ * Selector used to get the SSRC-rewriting feature flag.
+ *
+ * @param {Object} state - The global state.
+ * @returns {boolean}
+ */
+export function getSsrcRewritingFeatureFlag(state: IState) {
+    return getFeatureFlag(state, FEATURE_FLAGS.SSRC_REWRITING);
```
Comment (jallamsetty1) [inline react/features/base/config/functions.any.ts:95]: Do we need test if source-name signaling is explicitly disabled ? ssrc-rewriting is works only with source-name signaling right ?

File: react/features/base/tracks/reducer.ts [hunk 3]
```diff
@@ -12,7 +18,8 @@ import {
     TRACK_REMOVED,
     TRACK_UPDATE_LAST_VIDEO_MEDIA_EVENT,
     TRACK_UPDATED,
-    TRACK_WILL_CREATE
+    TRACK_WILL_CREATE,
+    TRACK_OWNER_CHANGED
```
Comment (jallamsetty1) [inline react/features/base/tracks/reducer.ts:22]: nit: alpha sort

File: react/features/base/tracks/actions.ts [hunk 4]
```diff
@@ -591,11 +595,14 @@ export function trackVideoStarted(track: any) {
  * }}
  */
 export function trackVideoTypeChanged(track: any, videoType: VideoType) {
+    const m = videoType === VIDEO_TYPE.CAMERA ? MEDIA_TYPE.VIDEO : MEDIA_TYPE.SCREENSHARE;
```
Comment (jallamsetty1) [inline react/features/base/tracks/actions.ts:598]: The constant can be renamed to mediaType and property value shorthand syntax can be used.

File: react/features/base/tracks/reducer.ts [hunk 5]
```diff
@@ -144,9 +174,11 @@ export type ITracksState = ITrack[];
 ReducerRegistry.register<ITracksState>('features/base/tracks', (state = [], action): ITracksState => {
     switch (action.type) {
     case PARTICIPANT_ID_CHANGED:
+    case PARTICIPANT_LEFT:
     case TRACK_NO_DATA_FROM_SOURCE:
     case TRACK_UPDATE_LAST_VIDEO_MEDIA_EVENT:
     case TRACK_UPDATED:
+    case TRACK_OWNER_CHANGED:
```
Comment (jallamsetty1) [inline react/features/base/tracks/reducer.ts:181]: nit: alpha sort

File: react/features/large-video/actions.any.js [hunk 6]
```diff
@@ -2,6 +2,9 @@
 
 import type { Dispatch } from 'redux';
 
+import {
```
Comment (jallamsetty1) [inline react/features/large-video/actions.any.js:5]: This can be a single line.

**Comment by jallamsetty1** [comment]
Jenkins test this please.

File: react/features/base/config/functions.any.ts (hunk 2 above)
Comment (jqdrqgnq) [reply react/features/base/config/functions.any.ts]: ok, fixed.

File: react/features/base/tracks/reducer.ts (hunk 3 above)
Comment (jqdrqgnq) [reply react/features/base/tracks/reducer.ts]: fixed.

File: react/features/base/tracks/actions.ts (hunk 4 above)
Comment (jqdrqgnq) [reply react/features/base/tracks/actions.ts]: fixed.

File: react/features/large-video/actions.any.js (hunk 6 above)
Comment (jqdrqgnq) [reply react/features/large-video/actions.any.js:5]: fixed.

File: react/features/base/tracks/reducer.ts (hunk 5 above)
Comment (jqdrqgnq) [reply react/features/base/tracks/reducer.ts]: fixed.

**Pull request closed without merging**